lib/ui/profile/interactor/profile_bloc.dart

```diff
@@ -56,15 +63,17 @@ class ProfileBloc extends Bloc<ProfileEvent, ProfileState> {
       if (result.isValue) {
         emit(state.copyWith(pageState: PageState.success, profileData: result.asValue!.value));
       } else {
+        final error = result.asError!.error;
         // if loading fails, show saved user data.
         final profileData = ProfileData(
-          name: userData.userName,
+          name: userData.userName ?? '',
           account: userData.accountName,
         );
-        emit(state.copyWith(pageState: PageState.success, profileData: profileData));
+        emit(state.copyWith(
+            pageState: PageState.success, profileData: profileData, doesNotHaveProfile: error.isNotFoundError));
       }
     } else {

```

Comment:
this case should never happen - on import, we create user data in the shared prefs..